lexos/static/js/scripts_upload.js

```diff
@@ -1,197 +1,197 @@
+      UploadAndParseFile(f,fileSize)
       // loading progress bar
-      if (f.type == '') {
-        $('#progress').html('Loading Workspace')
-      } else {
-        var calculatedWidth = String(180 * numberOfFileDone / totalFiles) + 'px'
-        $('#progress').html(numberOfFileDone + ' of ' + totalFiles).css('color', '#3498DB')
-        $('#progress-bar').css({ 'width': calculatedWidth })
+      if (f.type === '') {
+        progress.html('Loading Workspace')
+      }
+      else {
+        let calculatedWidth = String(180 * numberOfFileDone / totalFiles) + 'px'
+        progress.html(numberOfFileDone + ' of ' + totalFiles).css('color', '#3498DB')
+        progressBar.css({ 'width': calculatedWidth })
         if (numberOfFileDone / totalFiles > 0.5) {
-          $('#progress').css('color', '#FFF')
+          progress.css('color', '#FFF')
         }
-        if (added == 1) {
-          $('#progress-bar').html('Complete!').css({ 'color': '#FFF', 'text-align': 'center', 'width': '175px', 'height': '20px' }).fadeOut(2000)
-          $('.fa-folder-open-o')[0].dataset.originalTitle = 'You have ' + numberOfFileDone + ' active document(s)'
-          $('.fa-folder-open-o').fadeIn(200)
+        if (added === 1) {
+          progressBar.html('Complete!').css({ 'color': '#FFF', 'text-align': 'center', 'width': '175px', 'height': '20px' }).fadeOut(2000)
+          const faFolderOpen = $('.fa-folder-open-o')
+          faFolderOpen[0].dataset.originalTitle = 'You have ' + numberOfFileDone + ' active document(s)'
+          faFolderOpen.fadeIn(200)
           $('#status').hide()
         }
       }
     }
-
-    $('#progress').html('Ready For Files To Upload').css('color', '#074178').delay(3000).show()
+    progress.html('Ready For Files To Upload').css('color', '#074178').delay(3000).show()
     $id('fileselect').value = '' // this allows the event to fire on "change" in chrome. the value property changing is the
     // normal trigger, for some reason firefox overwrote this with their own behavior.
   }
 
-  // upload and display file contents
-  function UploadAndParseFile(file) {
-    var filename = file.name.replace(/ /g, '_')
+  // initialize
+  function Init() {
+    let fileselect = $id('fileselect'),
+        filedrag = $id('dragndrop')
+
+    // file select
+    fileselect.addEventListener('change', FileSelectHandler, false)
+
+    // is XHR2 available?
+    let xhr = new XMLHttpRequest()
+    if (xhr.upload) {
+      // file drop
+      filedrag.addEventListener('dragover', fileDragHover, false)
+      filedrag.addEventListener('dragleave', fileDragHover, false)
+      filedrag.addEventListener('drop', FileSelectHandler, false)
+    }
+  }
+
+  //Call initialization function
+  if (window.File && window.FileList && window.FileReader) {
+    Init()
+  }
+})
+
 
-    if (AllowedFileType(file.name) && file.size <= $id('MAX_FILE_SIZE').value) {
-      if (file.size == 0) {
+/**
+ * @return {boolean}
```

Comment:
This is not a documentation. You need to write the summary for this function and explain the parameter